### PR TITLE
Android version

### PR DESCRIPTION
**Pull request by dalinaum**
- [x] Logo View (Some bezier stuffs.)
- [x] Support themes.
- [x] Example activity.
- [x] Login activity
  - [x] UI
  - [x] Log in feauture
  - [x] Remember my account
  - [x] Form validation
  - [x] Return the result to callee Activity.
- [x] Register activity
  - [x] UI
  - [x] Register feature
  - [x] Form validation
  - [x] Remember my account
- [x] Support pre-defined Server URI
- [x] Support hiding Server URI

### CODE REVIEW

**Comment by dalinaum** [comment]
Please review this.

It doesn't include some icons such Server, Email and Password. Since they don't match Android's TextInputLayout.

screenshot-2017-04-04_21 47 07 588
screenshot-2017-04-04_21 47 21 300
screenshot-2017-04-04_21 47 32 306
screenshot-2017-04-04_21 47 47 978
screenshot-2017-04-04_21 48 08 738

File: RealmLoginKit Android/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bmeike) [inline RealmLoginKit Android/.idea/compiler.xml]: Do you really want to check in the .idea directory?

File: RealmLoginKit Android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (bmeike) [inline RealmLoginKit Android/app/build.gradle]: Are there actually any jars?  Would it make sense to remove this dependency?

File: RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            public void run() {
+                resizeLogo();
+            }
+        }, 100);
+    }
+
+    private void resizeLogo() {
+        final int width = getWindow().getDecorView().getWidth();
+        final int height = getWindow().getDecorView().getHeight();
+        final int minWidth;
```
Comment (bmeike) [inline RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java]: perhaps:
final int minWidth = (int) (((width < height) ? width : height) * LOGO_RATION)

File: RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            LoginKit.loginKit(this)
+                    .setDarkMode(isDarkMode)
+                    .setAppTitle("Example App")
+                    .setServerUri("127.0.0.7", false)
+                    .logIn();
+        }
+    }
+
+    private void handleThemeIfNeeded(int viewId) {
+        if (viewId == R.id.light_button) {
```
Comment (bmeike) [inline RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java]: Perhaps this should be a switch?  If R.id.light_button is a constant (this is not a library) that will work.

File: RealmLoginKit Android/app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<resources>
```
Comment (bmeike) [inline RealmLoginKit Android/app/src/main/res/values/dimens.xml:1]: I don't see how to comment on those .png files.  Shouldn't we use our logo or something, for the launcher icon?

**Review by bmeike — APPROVED** [review]
I'm have a couple questions, but nothing that blocks.

File: RealmLoginKit Android/.idea/copyright/profiles_settings.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
```
Comment (zaki50) [inline RealmLoginKit Android/.idea/copyright/profiles_settings.xml]: Do we need to include some files in `.idea/` directory?
If there is no particular reason, we should use the same `.gitignore` with other demo apps or `raelm-java`

**Review by zaki50 — CHANGES_REQUESTED** [review]

File: RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java (hunk 3 above)
Comment (zaki50) [reply RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java]: Or `Math.min(int,int)`

Comment (dalinaum) [reply RealmLoginKit Android/app/src/main/java/io/realm/realmloginkit/example/ExampleActivity.java]: OK. I will use mixed solution.

File: RealmLoginKit Android/app/src/main/res/values/dimens.xml (hunk 5 above)
Comment (dalinaum) [reply RealmLoginKit Android/app/src/main/res/values/dimens.xml:1]: I agree with you. I will make a launcher icon.

Comment (dalinaum) [reply RealmLoginKit Android/app/src/main/res/values/dimens.xml:1]: @bmeike I updated a launcher icon.

File: RealmLoginKit Android/.idea/copyright/profiles_settings.xml (hunk 6 above)
Comment (dalinaum) [reply RealmLoginKit Android/.idea/copyright/profiles_settings.xml]: @zaki50 Removed the entire `.idea` and updated `.gitignore`.

**Comment by beeender** [comment]
How easy to integrate this into user's app project? Need something about installation to the `README`?

**Review by TimOliver — APPROVED** [review]

**Comment by dalinaum** [comment]
@beeender Of course, README should be updated. Although it is not delivered by J Center or other repositories, I don't know how can users install this library easily.

**Comment by TimOliver** [comment]
I just downloaded and deployed it to one of the office Pixel tablets. It looks and behaves really nicely!

I think I found one small glitch. If you're holding the tablet in sideways, the keyboard overlays the password field, and there doesn't seem to be a way to scroll to it.

That's a very minor edge case, so I'm happy to merge this now and address it later.

Does anyone else think there are any outstanding issues that absolutely must be solved before we can go ahead and merge? :)

Thanks!

**Comment by dalinaum** [comment]
Althought README.md should be updated and its layout should be scrolled, I think they are not big issues and they can be another PRs.

**Review by beeender — APPROVED** [review]
👍

**Pull request merged**